asf_heat_pump_suitability/getters/load_tree_input.py

```
@@ -34,22 +42,58 @@ def load_gdf_os_openmap_local_layer(
         'tidal_boundary',
         'tidal_water',
         'woodland'
+
     Returns:
         gpd.GeoDataFrame: OS OpenMap Local geometries for specified layer
     """
     if not grid_squares:
-        print(f"\nLoading OS OpenMap Local - {layer.title()}...")
         print(f"Loading OS OpenMap Local - {layer.title()}...")
         return gpd.read_file(
-            config["data"]["geodata"]["gb_os_openmap_local"], layer=layer, **kwargs
-        )
+            filename=config["data"]["geodata"]["gb_os_openmap_local"],
+            layer=layer,
+            **kwargs,
+        ).drop_duplicates(subset="ID")
+
     else:
+        if not isinstance(grid_squares, List):
+            grid_squares = [grid_squares]
+        # Reformat layer name to how it appears in file name
         layer = layer.replace("_", " ").title().replace(" ", "")
         file_path = config["data"]["geodata"]["grid_square_os_openmap_local"]
         files = [file_path.format(square=code, layer=layer) for code in grid_squares]
+
         gdfs = []
+
         for file in files:
             print(f"\nLoading OS OpenMap Local - {layer.title()} file: {file}")
             gdfs.append(gpd.read_file(file, **kwargs))
 
-        return pd.concat(gdfs)
+        return pd.concat(gdfs).drop_duplicates(subset="ID")
+
+
+def load_gdf_poi() -> gpd.GeoDataFrame:
+    """
+    Load and process Points of Interest data. CRS EPSG 4326.
+
+    Returns:
+        gpd.GeoDataFrame: Processed POI data containing types of POI specified
+
+    Raises:
+        ValueError: If required columns are missing
+    """
+    print("Loading POI data...")
+
+    required_columns = [
+        "id",
+        "country",
+        "main_category",
+        "alternate_category",
+        "geometry",
+    ]
+    poi = gpd.read_file(
+        filename=config["data_source"]["UK_poi_locations"],
+        columns=required_columns,
+        layer="poi_uk",
+    ).to_crs("EPSG:4326")
```

Apologies for my ignorance on this, but is it problematic that we're using CRS 27700 and EPSG 4326 here?

---

Thank you for flagging this - I double checked the code and I think it's all good but good to check.

The CRS here is WGS84 (which is the same as EPSG 4326) - i.e. the coordinates are latitude and longitude. We have loaded the dataset in the WGS84 CRS here because that's the CRS of the original file.

In `asf_heat_pump_suitability/pipeline/transform/poi.py` in the `transform_gdf_poi` function, we convert it to the target CRS for this project which is 27700. [See line here](https://github.com/nestauk/asf_heat_pump_suitability/blob/9576f7ed9193bff92d82ef2ed5127a5a854a7bb5/asf_heat_pump_suitability/pipeline/transform/poi.py#L38). So whenever we conduct any joins or use the POI data, it will be transformed to 27700 first and then used.

But otherwise, you are correct, if we were mixing two different CRS then we would get silent failure in our spatial joins.